### DISCUSSION TITLE
mj-include

### DISCUSSION THREAD

**Discussion by jkuryanal** [SebastianStehle/mjml-net#137] [General]
Hi, we using mj-include extensively for layouts and css. Is there any plans if near future to add mj-include functionality to the code.?

**Comment by SebastianStehle** [comment]
Not at the moment, but PRs are very welcome.

I use mjml only in a server context, where it is not really needed. You always make a simple string concatenation before you call render. MJML seems to do the same.

https://documentation.mjml.io/#mj-include

> The MJML engine will then replace your included files before starting the rendering process.

**Comment by LiamRiddell** [comment]
@jkuryanal How complicated are the MJML templates that you're including with mj-include? For example, does it use `<mj-head>` extensively?

The reason I ask, is because as a potential workaround you could pre-process the MJML template using regex to find matches `<mj-include />` and then handle loading the file and insert in replacement of the `<mj-include>`.  However, if you need the full fledge functionality of `mj-include` like style merging then PR's are welcome.

**Reply by SebastianStehle** [reply]
I think it does not matter, as far as I understand the javascript implementation they make a preprocessing as well. There is no style merging, if I understand it properly

**Comment by SebastianStehle** [comment]
I have created a PR: https://github.com/SebastianStehle/mjml-net/pull/139

The good thing: it does not need preprocessing.

**Comment by SebastianStehle** [comment]
Implemented now.
